reliability/core-it/src/androidTest/kotlin/com/datadog/android/core/integration/tests/FeatureScopeTest.kt

```diff
@@ -0,0 +1,206 @@
+            val eventMetadata = forge.anAlphabeticalString()
+            RawBatchEvent(
+                fakeEvent.toString().toByteArray(),
+                eventMetadata.toByteArray()
+            )
+        }
+        fakeBatchMetadata = forge.anAlphabeticalString().toByteArray()
+        eventType = forge.aValueFrom(EventType::class.java)
+        stubFeature = StubStorageBackedFeature(
+            forge,
+            fakeFeatureName,
+            getMockServerWrapper().getServerUrl()
+        )
+        fakeConfiguration = forge.getForgery()
+    }
+
+    @After
+    fun tearDown() {
+        cleanupStorage()
+        Datadog.stopInstance()
+        cleanMockWebServer()
+    }
+
+    // region Consent Granted
+
+    @Test
+    fun mustWriteTheEvents_whenFeatureWrite_trackingConsentGranted() {
+        // Given
+        trackingConsent = TrackingConsent.GRANTED
+        testedInternalSdkCore = Datadog.initialize(
+            context = ApplicationProvider.getApplicationContext(),
+            configuration = fakeConfiguration,
+            trackingConsent = trackingConsent
+        ) as InternalSdkCore
+        testedInternalSdkCore.registerFeature(stubFeature)
+        val featureScope = testedInternalSdkCore.getFeature(fakeFeatureName)
+
+        // When
+        checkNotNull(featureScope)
+        featureScope.withWriteContext { _, eventBatchWriter ->
+            fakeBatchData.forEach { rawBatchEvent ->
+                eventBatchWriter.write(
+                    rawBatchEvent,
+                    fakeBatchMetadata,
+                    eventType
+                )
+            }
+        }
+
+        // Then
+        ConditionWatcher {
+            MockWebServerAssert.assertThat(getMockServerWrapper())
+                .withConfiguration(fakeConfiguration)
+                .withTrackingConsent(trackingConsent)
+                .receivedData(fakeBatchData, fakeBatchMetadata)
+            true
+        }.doWait(LONG_WAIT_MS)
+    }
+
+    // endregion
+
+    // region Consent Pending
+
+    @Test
+    fun mustNotWriteTheEvents_whenFeatureWrite_trackingConsentPending() {
+        // Given
+        val trackingConsent = TrackingConsent.PENDING
+        testedInternalSdkCore = Datadog.initialize(
+            context = ApplicationProvider.getApplicationContext(),
+            configuration = fakeConfiguration,
+            trackingConsent = trackingConsent
+        ) as InternalSdkCore
+        testedInternalSdkCore.registerFeature(stubFeature)
+        val featureScope = testedInternalSdkCore.getFeature(fakeFeatureName)
+
+        // When
+        checkNotNull(featureScope)
+        featureScope.withWriteContext { _, eventBatchWriter ->
+            fakeBatchData.forEach { rawBatchEvent ->
+                eventBatchWriter.write(
+                    rawBatchEvent,
+                    fakeBatchMetadata,
+                    eventType
+                )
+            }
+        }
+
+        // Then
+        Thread.sleep(MEDIUM_WAIT_MS)
```

Comment:
Why Thread.sleep here instead of `.doWait(LONG_WAIT_MS)`?

Reply:
because if you use condition here it will always be true as first time you check the MockWebServer will always have 0 requests. In this case we need to wait first and then check.